integ-tests/src/commonTest/kotlin/dev/evo/elasticmagic/ParentChildTests.kt

```diff
@@ -143,6 +146,67 @@ class ParentChildTests : ElasticsearchTestBase() {
                     val parentsAgg = searchResult.agg<TermsAggResult<String>>("parents")
                     parentsAgg.buckets shouldBe listOf(TermBucket("q~1", docCount = 3))
                 }
+
+            SearchQuery(sourceFactory)
+                .filter(HasParent(QuestionDoc.text.match("What"), "question"))
+                .execute(index)
+                .let { searchResult ->
+                    searchResult.hits.toSet() shouldContainExactly setOf(
+                        SearchHit(
+                            index = index.name,
+                            type = "_doc",
+                            id = "a~1",
+                            routing = "q~1",
+                            score = 0.0F,
+                            source = a1,
+                        ),
+                        SearchHit(
+                            index = index.name,
+                            type = "_doc",
+                            id = "a~2",
+                            routing = "q~1",
+                            score = 0.0F,
+                            source = a2,
+                        ),
+                    )
+                }
+
+            SearchQuery(sourceFactory)
+                .filter(HasParent(QuestionDoc.text.match("Nothing"), "question"))
+                .execute(index)
+                .let { searchResult ->
+                    searchResult.hits.size shouldBe 0
+                }
+
+            SearchQuery(sourceFactory)
+                .filter(HasChild(MatchAll, "answer", minChildren = 1))
+                .execute(index)
+                .let { searchResult ->
+                    searchResult.hits.toSet() shouldContainExactly setOf(
+                        SearchHit(
+                            index = index.name,
+                            type = "_doc",
+                            id = "q~1",
+                            routing = "q~1",
+                            score = 0.0f,
+                            source = q1,
+                        )
+                    )
+                }
+
+            SearchQuery(sourceFactory)
+                .filter(HasChild(MatchAll, "answer", minChildren = 3))
+                .execute(index)
+                .let { searchResult ->
+                    searchResult.hits.size shouldBe 0
+                }
+
+            SearchQuery(sourceFactory)
+                .filter(HasChild(MatchAll, "answer", maxChildren = 1))
+                .execute(index)
+                .let { searchResult ->
+                    searchResult.hits.size shouldBe 0
+                }
```

Comment:
I think we should make queries that return some documents. It is hard to ensure that query expression works as expected if there are no hits in a result